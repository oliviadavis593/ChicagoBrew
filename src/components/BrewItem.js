import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import '../styles/BrewItem.css';


class BrewItem extends Component {
    render() {
        const { image, name, id } = this.props; 

        return(
            <div className='brew-item'>
                <div className='brew-image'>
                    <img src={image} />
                </div>
                <div>
                  <Link to={`/brews/${id}`}>
                    <p className='gbt-item__brew_name'>
                        {name}
                    </p>
                  </Link>
                </div>
                <div>

                </div>
            </div>
        )
    }
}

export default BrewItem; 