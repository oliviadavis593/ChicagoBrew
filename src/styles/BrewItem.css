.brew-item {
    --item-height: 60px;
    background-color: #eee;
    padding: 8px 0px 8px 8px; 
    margin: 0 20px 15px -20px;
    box-shadow: 0 2px 4px 0 rgba(0,0,0,0.4); 
    width: 250px; 
    margin: 30px auto; 
}

.brew-image img{
    width: 100px;
    margin: 0 auto;  
}