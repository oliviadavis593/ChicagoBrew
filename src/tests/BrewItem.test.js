import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import BrewItem from '../components/BrewItem';

describe('BrewItem component', () => {
    it (`renders without crashing`, () => {
        const div = document.createElement('div')
        ReactDOM.render(<BrowserRouter><BrewItem name='test-name'/></BrowserRouter>, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it('renders the UI as expected', () => {
        const tree = renderer
            .create(<BrowserRouter><BrewItem  name='test-name'/></BrowserRouter>)
            .toJSON();
        expect(tree).toMatchSnapshot();
        })
})