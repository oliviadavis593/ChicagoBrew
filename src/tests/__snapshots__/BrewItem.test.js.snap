// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrewItem component renders the UI as expected 1`] = `
<div
  className="brew-item"
>
  <div>
    <a
      href="/brew/undefined"
      onClick={[Function]}
    >
      <p
        className="gbt-item__brew_name"
      >
        test-name
      </p>
    </a>
  </div>
</div>
`;
